app/components/post_body/post_body.js

```diff
@@ -400,6 +400,38 @@ export default class PostBody extends PureComponent {
                     theme={theme}
                 />
             );
+        } else if (postType === Posts.POST_TYPES.PURPOSE_CHANGE || postType === Posts.POST_TYPES.HEADER_CHANGE ||
+            postType === Posts.POST_TYPES.DISPLAYNAME_CHANGE || postType === Posts.POST_TYPES.CHANNEL_DELETED) {
+            const username = this.props.post.props['username'];
+            const prefixUsername = (announcement, name) => {
```

Comment:
The proper way to handle this, would be something similar to the way it was/is done for the web app. However, currently, the code utilizes `<CombinedSystemMessage />` component for where is it should be either replaced by a more general helper, or have the above "annoucements" integrated into `<CombinedSystemMessage />. I offered the latter to @enahum and if I understood Elias correctly, he suggested that `CombinedSystemMessage` better off being refactored at a later stage.

This should also somewhat answer the question above - `messageComponent` currently in similar to the generic case. The duplication of the few lines of code in question could be easily avoided, for example by checking the post type inside `if (message.length > 0)` block. However, given that this is just a temporary solution, I am unsure it makes sense. Some of the code clarity would be lost, and during refactoring, the developer will probably want to break the post types back and either use a generic helper, possibly with internationalization, or have a special "helped" to handle the four types of posts in question.

Reply:
@igorshukler I'm totally fine to have `CombinedSystemMessage` component be the one responsible for doing anything that is related to a system message, you would need to rename the component to be something more obvious as I don't think `Combined` would continue to work.

Actually this is my proposal:
Create a HOC component for SystemMessages and do handle the different SystemMessages types in here and `CombinedSystemMessages` could be renamed to `CombinedJoinLeaveMessages`.

thoughts?

Reply:
Sure, makes sense to me.

Reply:
I have spent a bit of time on your suggestion @enahum - wrapping a HOC. I actually cannot figure out why/how this would be useful in our case. Typically, HOCs are made to reuse functionality. What functionality were you thinking to reuse between the `combined` message component and one for the four (4) announcements in question?

Reply:
So high level this is what I was thinking.

```jsx
<SystemMessage>
   {if is combined <CombinedJoinLeaveMessages>}
   { if case 1 <SystemMessageCase1>}
  { if case 2 <SystemMessageCase2>}
  {...}
</SystemMessage>
```

does it make sense to you?

Reply:
@enahum
Sketched how I understood what you are suggesting. Pushed the updated code.